pennylane/ops/qubit/matrix_ops.py

```diff
@@ -296,9 +319,92 @@ def compute_decomposition(U: TensorLike, wires: WiresLike):
                 raise DecompositionUndefinedError(
                     "The decomposition of a two-qubit sparse QubitUnitary is undefined."
                 )
+
             return qml.ops.two_qubit_decomposition(U, Wires(wires))
 
-        return super(QubitUnitary, QubitUnitary).compute_decomposition(U, wires=wires)
+        ops = []
+
+        # pylint: disable=import-outside-toplevel
+        try:
```

Comment:
Thanks a lot for the excellent work! Just a minor suggestion that would you like to separate the arbitrary-n qubit unitary decomposition out of `compute_decomposition`? Like just have another inplace method named `multi_qubit_decomposition` or whatever other proper names; this will make things slightly clearer here and help future developers to quickly locate whenever anything needs maintenance.

Reply:
This suggestion can be covered by another: https://github.com/PennyLaneAI/pennylane/pull/7277#discussion_r2080147743
So resolved